components/measurement/FeedbackBox.js

```diff
@@ -0,0 +1,254 @@
+        }
+      },
+      blocking: {
+        on: {
+          CANCEL: 'initial',
+          SUBMIT: 'submit'
+        }
+      },
+      notBlocking: {
+        on: {
+          CANCEL: 'initial',
+          SUBMIT: 'submit'
+        }
+      },
+      closed: {},
+      submit: {
+        effect({ send, setContext, event, context }) {
+          const { nested, feedback } = getValues()
+          const feedbackParams = {
+            measurement_uid: report_id,
+            status: nested || feedback
+          }
+          submitFeedback(feedbackParams)
+            .then(() => send('SUCCESS'))
+            .catch(({response}) => {
+              setError(response?.data?.error || 'unknown')
+              send('FAILURE')
+            })
+        },
+        on: {
+          SUCCESS: 'success',
+          FAILURE: 'failure'
+        }
+      },
+      success: {
+        on: {
+          CLOSE: 'closed'
+        }
+      },
+      failure: {
+        on: {
+          CLOSE: 'closed',
+          SUBMIT: 'submit'
+        },
+        effect() {
+          return () => setError(null)
+        },
+      }
+    },
+  })
+
+  const firstLevel = watch('feedback')
+  const nestedLevel = watch('nested')
+ 
+  useEffect(() => {
+    setValue('nested', null)
+  }, [firstLevel, setValue])
+
+  const sumitEnabled = useMemo(() => firstLevel || nestedLevel, [firstLevel, nestedLevel])
+
+  return (
+    <>
+      {state.value !== 'closed' && 
+        <Box
+          p={3}
+          bg='gray3'
+          sx={{
+            width: '400px',
+            position: 'absolute',
+            right: '60px',
+            background: 'silver',
+            borderRadius: '10px'
+          }}
+        >
+          <form onSubmit={(e) => e.preventDefault()}>
+            {state.value === 'initial' && 
+              <>
+                {user.loggedIn ? 
+                  <>
+                    {previousFeedback ? 
+                      <>
+                        <Text fontSize={16} mb={3}><FormattedMessage id="Measurement.Feedback.ExistingFeedback" /></Text>
+                        <Text fontSize={16} mb={3}><FormattedMessage id={`Measurement.Feedback.${previousFeedback}`} /></Text>
+                        <Button mr={2} onClick={() => send('IS_BLOCKING')}>
+                          <FormattedMessage id="General.Edit" />
+                        </Button>
+                      </> :
+                      <>
+                        <QuestionText i18nKey='Measurement.Feedback.Q1' />
+                        <Button mr={2} onClick={() => send('IS_BLOCKING')}><FormattedMessage id='General.Yes' /></Button>
+                        <Button onClick={() => send('CLOSE')}><FormattedMessage id='General.No' /></Button>
+                      </>
+                    }
+                  </>
+                  : 
+                  <Text fontSize={16}>
+                    <FormattedMessage
+                      id='Measurement.Feedback.Login'
```

Comment:
I think that we should only prompt the user to login once they have expressed interest in submitting feedback.

That is to say we should present the same initial state irrespective of the logged in state and only on the second step, only if they are not logged in, ask them to login/register to provide feedback.